server/src/main/java/org/elasticsearch/search/fetch/FetchPhase.java

```diff
@@ -207,13 +207,17 @@ private SearchHit createSearchHit(SearchContext context,
         DocumentMapper documentMapper = context.mapperService().documentMapper();
         Text typeText = documentMapper.typeText();
         if (fieldsVisitor == null) {
-            return new SearchHit(docId, null, typeText, null);
+            return new SearchHit(docId, null, typeText, null, null);
         }
 
         Map<String, DocumentField> searchFields = getSearchFields(context, fieldsVisitor, subDocId,
             storedToRequestedFields, subReaderContext);
+        
+        Map<String, DocumentField> metaFields = new HashMap<>(), 
+                documentFields = new HashMap<>();
```

Comment:
nit: Please use a separate line for declaration of documentFields, no need for a continuation of the type.